lab-nassir/lib/readFiles.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+// node modules
+const fs = require('fs');
+// npm modules
+// app modules
+// module constants
+// module logic
+
+module.exports = function readDataFiles(callback) {
+  var resultArray = [];
+  fs.readFile(`${__dirname}/../data/one.txt`, function(err, data){
+    resultArray.push(data.toString('hex', 0, 7));
+    fs.readFile(`${__dirname}/../data/two.txt`, function(err, data){
+      resultArray.push(data.toString('hex', 0, 7));
+      fs.readFile(`${__dirname}/../data/three.txt`, function(err, data){
+        resultArray.push(data.toString('hex', 0, 7));
```

Comment:
If you read about the toString method, the third argument is the end point (non-inclusive). https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end Therefore, you should be invoking toString like so to get the first **8 bytes:**

`data.toString('hex', 0, 8)`